### PR TITLE
swift-format experiments

### PR DESCRIPTION
**Pull request by adam-fowler**
Experimenting with moving to swift-format
So far I copied the .swift-format from swift-nio and ran swift format on the Tests and Sources folders

### CODE REVIEW

**Comment by adam-fowler** [comment]
Running `swift format lint` brings up a few issues as well
```
Sources/HummingbirdCore/Utils/HBParser.swift:677:21: warning: [NoBlockComments] replace this block comment with line comments
/Users/adamfowler/Developer/server/hummingbird-project/hummingbird/Sources/HummingbirdCore/Utils/HBParser.swift:61:113: warning: [EndOfLineComment] move end-of-line comment that exceeds the line length
Sources/Hummingbird/Environment.swift:140:17: warning: [OmitExplicitReturns] 'return' can be omitted because body consists of a single expression
Sources/Hummingbird/Codable/URLEncodedForm/URLEncodedFormNode.swift:31:19: warning: [ReplaceForEachWithForLoop] replace use of '.forEach { ... }' with for-in loop
Sources/Hummingbird/Codable/URLEncodedForm/URLEncodedFormNode.swift:179:36: warning: [OneCasePerLine] move 'map' to its own 'case' declaration
Sources/Hummingbird/Codable/URLEncodedForm/URLEncodedFormNode.swift:179:59: warning: [OneCasePerLine] move 'arrayWithIndices' to its own 'case' declaration
Sources/Hummingbird/Server/RequestID.swift:39:24: warning: [DontRepeatTypeInStaticProperties] remove the suffix 'RequestID' from the name of the variable 'globalRequestID'
Sources/Hummingbird/HTTP/MediaType.swift:400:16: warning: [DontRepeatTypeInStaticProperties] remove the suffix 'MediaType' from the name of the variable 'extensionMediaTypeMap'
Sources/Hummingbird/Router/Router.swift:60:23: warning: [ReplaceForEachWithForLoop] replace use of '.forEach { ... }' with for-in loop
```

File: Sources/Hummingbird/Codable/JSON/JSONCoding.swift [hunk 1]
```diff
@@ -12,18 +12,20 @@
...
 
 extension JSONEncoder: ResponseEncoder {
     /// Extend JSONEncoder to support encoding `Response`'s. Sets body and header values
     /// - Parameters:
     ///   - value: Value to encode
     ///   - request: Request used to generate response
     ///   - context: Request context
-    public func encode(_ value: some Encodable, from request: Request, context: some RequestContext) throws -> Response {
+    public func encode(_ value: some Encodable, from request: Request, context: some RequestContext) throws -> Response
+    {
```
Comment (Joannis) [inline Sources/Hummingbird/Codable/JSON/JSONCoding.swift:28]: This type of formatting is super annoying, is that configurable?

File: Sources/Hummingbird/Codable/URLEncodedForm/URLEncodedForm+Request.swift [hunk 2]
```diff
@@ -18,7 +18,8 @@ extension URLEncodedFormEncoder: ResponseEncoder {
     ///   - value: Value to encode
     ///   - request: Request used to generate response
     ///   - context: Request context
-    public func encode(_ value: some Encodable, from request: Request, context: some RequestContext) throws -> Response {
+    public func encode(_ value: some Encodable, from request: Request, context: some RequestContext) throws -> Response
+    {
```
Comment (Joannis) [inline Sources/Hummingbird/Codable/URLEncodedForm/URLEncodedForm+Request.swift:22]: "

**Review by Joannis — APPROVED** [review]
I like this a lot, just one thing that's annoying IMO. But I can live with it if it's not fixable

File: Sources/Hummingbird/Codable/JSON/JSONCoding.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/Hummingbird/Codable/JSON/JSONCoding.swift:28]: Yeah I haven't worked out how to stop that. Not very keen on it

File: Tests/HummingbirdRouterTests/RouterTests.swift [hunk 3]
```diff
@@ -310,7 +334,8 @@ final class RouterTests: XCTestCase {
         }
         let app = Application(responder: router)
         try await app.test(.router) { client in
-            try await client.execute(uri: "/test/group", method: .get, headers: [.middleware2: "Transforming"]) { response in
+            try await client.execute(uri: "/test/group", method: .get, headers: [.middleware2: "Transforming"]) {
+                response in
```
Comment (adam-fowler) [inline Tests/HummingbirdRouterTests/RouterTests.swift:338]: @Joannis this is pretty horrible as well

**Pull request closed without merging**